r/vignettes/dataset.Rmd

```diff
@@ -77,39 +79,44 @@ feel free to grab only a year or two of data.
 
 If you don't have the taxi data downloaded, the vignette will still run and will
 yield previously cached output for reference. To be explicit about which version
-is running, let's check whether we're running with live data:
+is running, let's check whether you're running with live data:
 
 ```{r}
 dir.exists("nyc-taxi")
 ```
 
-## Getting started
+## Opening the dataset
 
-Because `dplyr` is not necessary for many Arrow workflows,
+Because dplyr is not necessary for many Arrow workflows,
 it is an optional (`Suggests`) dependency. So, to work with Datasets,
-we need to load both `arrow` and `dplyr`.
+you need to load both arrow and dplyr.
 
 ```{r}
 library(arrow, warn.conflicts = FALSE)
 library(dplyr, warn.conflicts = FALSE)
 ```
 
-The first step is to create our Dataset object, pointing at the directory of data.
+The first step is to create a Dataset object, pointing at the directory of data.
 
 ```{r, eval = file.exists("nyc-taxi")}
 ds <- open_dataset("nyc-taxi", partitioning = c("year", "month"))
 ```
 
-The default file format for `open_dataset()` is Parquet; if we had a directory
-of Arrow format files, we could include `format = "arrow"` in the call.
-Other supported formats include: `"feather"` (an alias for `"arrow"`, as Feather
-v2 is the Arrow file format), `"csv"`, `"tsv"` (for tab-delimited), and `"text"`
-for generic text-delimited files. For text files, you can pass any parsing
-options (`delim`, `quote`, etc.) to `open_dataset()` that you would otherwise
-pass to `read_csv_arrow()`.
+The file format for `open_dataset()` is controlled by the `format` parameter, 
+which has a default value of `"parquet"`.  If you had a directory
+of Arrow format files, you could instead specify `format = "arrow"` in the call.
+
+Other supported formats include: 
+
+* `"feather"` or `"ipc"` (aliases for `"arrow"`, as Feather v2 is the Arrow file format)
+* `"csv"` (comma-delimited files) and `"tsv"` (tab-delimited files)
+* `"text"` (generic text-delimited files - use the `delimiter` argument to specify which to use)
 
-The `partitioning` argument lets us specify how the file paths provide information
-about how the dataset is chunked into different files. Our files in this example
+For text files, you can pass any parsing options (`delim`, `quote`, etc.) to 
+`open_dataset()` that you would otherwise pass to `read_csv_arrow()`.
```

Comment:
I have no recollection of that but I'll have a search on JIRA and see what I can find

Reply:
Had a look in the docs and I found this in `open_dataset()`: 
> additional arguments passed to dataset_factory() when sources is a directory path/URI or vector of file paths/URIs

Is this what you meant, as in it only works for certain values of `sources`, or something else?

Reply:
I was thinking about https://github.com/apache/arrow/blob/master/r/R/dataset-format.R#L135-L153 bit. 

Running the code in the middle of that, I get:

```
>   # Catch any readr-style options specified with full option names that are
>   # supported by read_delim_arrow() (and its wrappers) but are not yet
>   # supported here
>   unsup_readr_opts <- setdiff(
+     names(formals(read_delim_arrow)),
+     names(formals(readr_to_csv_parse_options))
+   )
> unsup_readr_opts
 [1] "file"              "schema"            "col_names"         "col_types"         "col_select"       
 [6] "na"                "quoted_na"         "skip"              "parse_options"     "convert_options"  
[11] "read_options"      "as_data_frame"     "timestamp_parsers"
```

Reply:
you may want to intersect that with `names(formals(readr::read_delim))` since some of those are arrow function args

Reply:
I'm confused by @nealrichardson 's comment, please can you rephrase that?

In the meantime, it looks like there are only 5 parsing options supported and more unsupported, so I feel like I'm better off explicitly listing the ones that are supported.

Reply:
Many of the args in `unsup_readr_opts` aren't actually from readr, they're just arguments to read_delim_arrow that aren't in readr_to_csv_parse_options. We only care about the ones that are readr options, so:

```
> intersect(unsup_readr_opts, names(formals(readr::read_delim)))
[1] "file"       "col_names"  "col_types"  "col_select" "na"        
[6] "quoted_na"  "skip" 
```

Reply:
Right, I get you, I think. I think that now I'm explicitly specifying the arguments that *can* be passed through rather than *can't*, I don't think I need to make any more changes here on account of the above comments?